app.tsx

```diff
@@ -47,20 +49,40 @@ const App = () => {
   const [loginMutation, { data, loading, error }] = useMutation(loginMutationGQL, { client: client });
 
   const handleButtonPress = () => {
+    loginErrorText.current = '';
     if (email.current == '' || password.current == '') {
-      setErrorMessage('Preencha todos os campos');
-    } else if (password.current.length < 7) {
-      setErrorMessage('A senha deve possuir pelo menos 7 caracteres');
-    } else if (!emailRegex.test(email.current)) {
-      setErrorMessage('O campo de email deve seguir o formato de um email: abc@def.com');
-    } else if (!passNumberRegex.test(password.current) || !passCharRegex.test(password.current)) {
-      setErrorMessage('O campo de senha deve conter ao menos uma letra e um número');
-    } else {
-      setErrorMessage('');
+      loginErrorText.current += 'Todos os campos devem ser preenchidos';
+    }
+    if (!emailRegex.test(email.current)) {
+      if (loginErrorText.current !== '') loginErrorText.current += '\n';
+      loginErrorText.current += 'O campo de email deve seguir o formato de um email: abc@def.com';
+    }
```

Comment:
vc poderia eliminar essas verificações do current adicionando o `\n` no final de cada uma das mensagens de erro
```suggestion
    if (!emailRegex.test(email.current)) {
      loginErrorText.current += 'O campo de email deve seguir o formato de um email: abc@def.com\n';
    }
```